src/link.js

```diff
@@ -113,11 +95,29 @@ module.exports = function link(config, args, callback) {
   };
 
   const tasks = dependencies.map((dependency) => (next) => {
-    const prelink = makeHook(dependency, 'prelink');
-    const postlink = makeHook(dependency, 'postlink');
-    const link = makeLink(dependency);
+    const queue = [];
+
+    var prelink;
+    var postlink;
+
+    if (dependency.config.hooks) {
+      prelink = dependency.config.hooks.prelink;
+      postlink = dependency.config.hooks.postlink;
+    }
+
+    if (prelink) {
+      queue.push(makeHook(prelink));
```

Comment:
I totally agree with it. That was a point I wanted to discuss with you yesterday when you went offline :+1: